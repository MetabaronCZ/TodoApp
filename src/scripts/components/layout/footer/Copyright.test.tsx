import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, it } from '@jest/globals';

import { withMockedProviders } from 'test/component';
import { Copyright } from 'components/layout/footer/Copyright';

describe('components/layout/Copyright', () => {
  it('should render correctly', () => {
    const tree = render(withMockedProviders(<Copyright />));
    expect(tree.container).toMatchSnapshot();

    const year = new Date().getFullYear();
    expect(tree.getByText(year, { exact: false })).toBeTruthy();
    expect(
      tree.container.querySelector('[href="https://github.com/MetabaronCZ"]'),
    ).toBeInTheDocument();
  });
});
