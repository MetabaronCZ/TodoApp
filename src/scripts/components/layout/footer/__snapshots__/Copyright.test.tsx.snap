// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/layout/Copyright > should render correctly 1`] = `
.c0 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #22c;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:focus {
  outline: 1px dotted #22c;
  outline-offset: -1px;
}

<div>
  <div
    class=""
  >
    Copyright Â© 2023 MetabaronCZ (
    <a
      class="c0"
      href="https://github.com/MetabaronCZ"
      target="_blank"
    >
      https://github.com/MetabaronCZ
    </a>
    ). All Rights Reserved.
  </div>
</div>
`;
